helix-term/src/ui/editor.rs

```diff
@@ -569,9 +572,22 @@ impl EditorView {
         cxt: &mut commands::Context,
         event: KeyEvent,
     ) -> Option<KeymapResult> {
+        self.autoinfo = None;
         match self.keymaps.get_mut(&mode).unwrap().get(event) {
             KeymapResult::Matched(command) => command.execute(cxt),
-            KeymapResult::Pending(_map) => { /* render infobox */ }
+            KeymapResult::Pending(map) => {
+                let mut body = BTreeMap::new();
+                for (key, trie) in map.into_iter() {
+                    let desc = match trie {
+                        KeyTrie::Leaf(cmd) => cmd.doc(),
+                        KeyTrie::Node(_) => "",
+                    };
+                    // FIXME: multiple keys are ordered randomly (use BTreeSet)
```

Comment:
The `keymaps` should preserve the order, it uses `HashMap` which does not.

Reply:
The items are ordered correctly now, I meant when the same function is bound to multiple keys (like `zc` and `zz`). It will be `c, z` or `z, c` but consistent in a  single editor session (changes when editor is closed and opened).